### PR TITLE
[AMORO-1744] Rename name with arctic identifier in amoro-mixed-format-spark module

### PR DESCRIPTION
**Pull request by zhoujinsong**
## Why are the changes needed?

As titled.

## Brief change log

- Change the name of classes with the arctic identifier.
- Change the name of variables and methods with the arctic identifier.
- Change the comments with the arctic identifier.
- Change related documentation.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by xleoken** [comment]
A big and great job~

File: amoro-mixed-format/amoro-mixed-format-spark/amoro-mixed-format-spark-3-common/src/test/java/org/apache/amoro/spark/test/MixedTableTestBase.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public void cleanUpSource() {
       return;
     }
     if (TestIdentifier.SOURCE_TYPE_ARCTIC.equalsIgnoreCase(source.sourceType)) {
```
Comment (XBaith) [inline amoro-mixed-format/amoro-mixed-format-spark/amoro-mixed-format-spark-3-common/src/test/java/org/apache/amoro/spark/test/MixedTableTestBase.java:102]: SOURCE_TYPE_MIXED?

Comment (zhoujinsong) [reply amoro-mixed-format/amoro-mixed-format-spark/amoro-mixed-format-spark-3-common/src/test/java/org/apache/amoro/spark/test/MixedTableTestBase.java:102]: The old `arctic` source is kept to be compatible with previous versions.
So are you suggesting changing the variable name to  SOURCE_TYPE_MIXED and keeping the variable value to arctic?

File: amoro-mixed-format/amoro-mixed-format-spark/amoro-mixed-format-spark-3-common/src/main/java/org/apache/amoro/spark/mixed/ArcticDataSource.java
Comment (baiyangtx) [inline amoro-mixed-format/amoro-mixed-format-spark/amoro-mixed-format-spark-3-common/src/main/java/org/apache/amoro/spark/mixed/ArcticDataSource.java:1]: Why not just delete the "arctic" data-source ?

File: amoro-mixed-format/amoro-mixed-format-spark/v3.2/amoro-mixed-format-spark-3.2/src/main/java/org/apache/amoro/spark/table/UnkeyedSparkTable.java [hunk 2]
```diff
@@ -64,6 +56,14 @@ public String name() {
         + unkeyedTable.id().getTableName();
   }
 
+  @Override
+  public Map<String, String> properties() {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.putAll(super.properties());
+    properties.put("provider", "arctic");
+    return properties;
+  }
```
Comment (baiyangtx) [inline amoro-mixed-format/amoro-mixed-format-spark/v3.2/amoro-mixed-format-spark-3.2/src/main/java/org/apache/amoro/spark/table/UnkeyedSparkTable.java:65]: "provider" may should be "mixed-iceberg" or "mixed-hive"

File: amoro-mixed-format/amoro-mixed-format-spark/amoro-mixed-format-spark-3-common/src/main/java/org/apache/amoro/spark/mixed/ArcticDataSource.java
Comment (zhoujinsong) [reply amoro-mixed-format/amoro-mixed-format-spark/amoro-mixed-format-spark-3-common/src/main/java/org/apache/amoro/spark/mixed/ArcticDataSource.java:1]: We may keep it to be compatible with previous versions and remove it in next major version.

File: amoro-mixed-format/amoro-mixed-format-spark/v3.2/amoro-mixed-format-spark-3.2/src/main/java/org/apache/amoro/spark/table/UnkeyedSparkTable.java (hunk 2 above)
Comment (zhoujinsong) [reply amoro-mixed-format/amoro-mixed-format-spark/v3.2/amoro-mixed-format-spark-3.2/src/main/java/org/apache/amoro/spark/table/UnkeyedSparkTable.java]: I have fixed it, PTAL.

**Review by baiyangtx — APPROVED** [review]
LGTM
